osrf/vrx
# Fix gymkhana scoring issues

See issue #317.

As described in the issue:

* I exposed the parameter `k` as an SDF parameter used in the station keeping scoring plugin. I named the parameter `heading_weight` but I'm not very convinced about it. Let me know if you can think of a better name.
* I added the logic to consider the WAM-V collisions to the gymkhana scoring plugin. We now have an extra SDF parameter to set the penalty for each collision. We now apply the penalties when the task finishes.

How to test it:

* The disable of the heading error is harder to test. The only quick way I can think of is to move closer to the black box, print the score continuously, and use the rotation tool to rotate the WAM-V verifying that there aren't big jumps in the score. Note that the scoring should change as the moving average keeps adjusting but we shouldn't see big jumps.

* For testing the obstacle penalty, you can drive the WAM-V through the channel,  then, hit one buoy,  print continuously the score and let the task finish. At the end you should see the correction. Here's an example:

```
---
name: "gymkhana"
state: "running"
ready_time: 
  secs: 10
  nsecs:         0
running_time: 
  secs: 20
  nsecs:         0
elapsed_time: 
  secs: 79
  nsecs:         0
remaining_time: 
  secs: 1
  nsecs:         0
timed_out: False
score: 93.83531551021375
---
name: "gymkhana"
state: "finished"
ready_time: 
  secs: 10
  nsecs:         0
running_time: 
  secs: 20
  nsecs:         0
elapsed_time: 
  secs: 80
  nsecs:         0
remaining_time: 
  secs: 0
  nsecs:         0
timed_out: True
score: 193.71494983741547
---
```

Note that for testing I modified the value of `running_state_duration` to `80` and `obstacle_penalty` to `100` to make it more evident.
